src/lib/rules/strict-transport-security/strict-transport-security.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @fileoverview Check if responses served over HTTPS also have the Strict-Transport-Security header with a proper value max-age value.
+ */
+
+import { RuleContext } from '../../rule-context'; // eslint-disable-line no-unused-vars
+import { isHTTPS, normalizeString } from '../../utils/misc'; // eslint-disable-line no-unused-vars
+import { IAsyncHTMLElement, IResponse, IFetchEnd, IRule, IRuleBuilder } from '../../types'; // eslint-disable-line no-unused-vars
+
+// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /** The minimum period (in seconds) allowed for `max-age`. */
+        let minMaxAgeValue: number;
+
+        /** match for each directive in the header */
+        type DirectiveMatch = { // eslint-disable-line no-unused-vars
+            'max-age': Array<string>,
+            'includesubdomains': Array<string>,
+            'preload': Array<string>
+        };
+
+        const loadRuleConfigs = () => {
+            minMaxAgeValue = (context.ruleOptions && context.ruleOptions.minMaxAgeValue) || 10886400; // 18 weeks
+        };
+
+        const parse = (headerValue: string): DirectiveMatch => {
+            const regexes = {
+                // `max-age=31536000; includeSubDomains; preload` => [`max-age=31536000`]
+                // `max-age=31536000; MAX-age=12345` => [`max-age=31536000`, `MAX-age=12345`]
+                includesubdomains: /includeSubDomains/gi,
+                'max-age': /max-age=([^;]+);|max-age=([^;]+)$/gi,
```

Comment:
From the looks of the code, `max-age=10886401,includesubdomains` passes this rule, and it should not. 

---

Maybe look at how a similar parser does it (e.g.: for [`content-type`](https://github.com/jshttp/content-type/blob/64bde0d996ccb4334341662c0c7d25f7b370c4d9/index.js)).